src/libraries/Common/src/Interop/Unix/System.Native/Interop.GetPageSize.cs

```diff
@@ -0,0 +1,15 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        internal static int PageSize { get => field == 0 ? (field = GetPageSize()) : field; }
```

Comment:
To make it thread-safe:
```suggestion
using System.Runtime.InteropServices;
using System.Threading;

internal static partial class Interop
{
    internal static partial class Sys
    {
        private static volatile int s_pageSize;

        internal static int PageSize
        {
            get
            {
                int size = s_pageSize;
                if (size == 0)
                {
                    Interlocked.CompareExchange(ref s_pageSize, GetPageSize(), 0);
                    size = s_pageSize;
                }

                return size;
            }
        }
```